src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckUtils.kt

```diff
@@ -172,6 +175,20 @@ fun AnnotationHolder.createAnnotationsForFile(file: RsFile, annotationResult: Rs
     }
 }
 
+fun Disposable.createChildDisposedOnAnyPsiChange(messageBus: MessageBus): Disposable {
+    val child = Disposable {}
+    Disposer.register(this, child)
+    messageBus.connect().subscribe(
```

Comment:
This is a memory leak: you register a listener and never unregister it. Looks like you should write `messageBus.connect(child)`

Reply:
Fixed.